### ISSUE TITLE
Markdown: Preserve non-breaking spaces in inline code

### ISSUE THREAD

**Issue by leafac** [prettier/prettier#6402]
Since https://github.com/prettier/prettier/issues/3320 non-breaking spaces are preserved in Markdown prose, but they still aren’t preserved in inline code, which would have been useful to avoid line breaks in things like `x = 2`.

In the listings below `␣` represents a regular space, and `•` represents a non-breaking space.

**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEUICcC2BDANgAgGcAHbMOAHSirSgFoAjdObAawEsoBzAApLMrUoAAzRY8RUuWFUqo6I2ZtOvftKogANCAjEY7aIWShs6dBADuABVMIjKbADcI7ACZaQTMqzgwAylIqyDDoAK5w2gAWMJi4AOqR7PBqcH52SeyOSQCeyOCERtqchHDoMFbo2Fw4yABmeCXaAFaEAB4AQpVgPv7YmHAAMpxwdQ0RIC2tfiq4cACKoRDwo7iNIKToJeh5OOisrpZQHsTonDBxbjCRyAAcAAzaJxAlcZXEeSdwW44j2gCOi3gFV09hA2EIdCgcDgrhhHmYAPYzAqVRqSHqq3GJUw7GCYSxM3mgJG6LG2hg2AYF1cV2QACZyZV2LgVABhCCYNGoaC-EChEoAFUp9gxJQAvmKgA)
```sh
--parser markdown
```

**Input:**
```markdown
normal␣space

non-breaking•space

`normal␣space`

`non-breaking•space`

```

**Output:**
```markdown
normal␣space

non-breaking•space

`normal␣space`

`non-breaking␣space`

```

**Expected behavior:**
```markdown
normal␣space

non-breaking•space

`normal␣space`

`non-breaking•space`

```

**Comment by ikatyang** [comment]
It seems we should only process normal whitespaces (0x20):

https://github.com/prettier/prettier/blob/23146404850011972f695fb6bc2b8113c3cffbfc/src/language-markdown/preprocess.js#L31-L41

**Comment by leafac** [comment]
I think we should leave inline code untouched, because we don’t know in which language it is so anything we do may break it. Suppose, for example, that I’m writing Markdown to document Prettier, and I want to write about how Prettier would convert multiple tabs.

**Comment by ikatyang** [comment]
According to the CommonMark [spec](https://spec.commonmark.org/0.29/#code-spans) _"Line endings are treated like spaces"_, newlines are always valid to process. As for the whitespaces and tabs, it is hard to decide if we should process them given that browsers _typically_ render them as single spaces, or should we respect the `htmlWhitespaceSensitivity` option in this case?

**Comment by leafac** [comment]
Yes, you’re right. Normal whitespaces (0x20), tabs (0x9), and newlines (0xA, 0xD) may all be processed. I don’t see a point in respecting `htmlWhitespaceSensitivity` in this case.

**Comment by ikatyang** [comment]
Those whitespaces are sensitive if you apply CSS (e.g., `code { white-space: pre; }`) on the rendered HTML.

**Comment by leafac** [comment]
If we were to consider that case, then we couldn’t change whitespace anywhere because of, for example, `p { white-space: pre; }`

**Comment by jellopuddingstick** [comment]
Just wanted to chime in that a fix for this would be great. My use case is aligning fixed-width text with spaces, e.g.
- `foo      bar`
- `longWord baz`

**Comment by andersk** [comment]
> According to the CommonMark [spec](https://spec.commonmark.org/0.29/#code-spans) _"Line endings are treated like spaces"_, newlines are always valid to process. As for the whitespaces and tabs, it is hard to decide if we should process them given that browsers _typically_ render them as single spaces, or should we respect the `htmlWhitespaceSensitivity` option in this case?

Whether or not newlines are significant to the _output_, replacing them with spaces can result in extremely long _source_ lines that make editing unpleasant. I opened #11372 for this.

**Comment by Xunnamius** [comment]
Also experiencing this. Being a two year old issue, I'm kinda surprised I'm only just now encountering it. Use case: I'm trying to write documentation for GitHub and don't want my inline code examples (e.g. `requestPatcher: (req) => (req.url = '/my-url?some=query')`) being broken up across multiple lines. Nothing a post-processing sed step can't fix in the meantime :)

**Comment by neodon** [comment]
Just ran into this issue.  Tried using a non-breaking space in some inline code in a table to keep it from wrapping.

Why does prettier mess with inline code at all?

**Comment by thorn0** [comment]
Fixed by #13590 in `next`
